src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -391,14 +397,63 @@ private IAIState canGoPlanting(@NotNull final ScarecrowTileEntity currentField,
      *
      * @param position the position to check.
      * @param field    the field close to this position.
-     * @return true if should be hoed.
+     * @return position of hoeable surface or null if there is not one
      */
-    private boolean shouldHoe(@NotNull final BlockPos position, @NotNull final ScarecrowTileEntity field)
+    private BlockPos findHoeableSurface(@NotNull BlockPos position, @NotNull final ScarecrowTileEntity field)
     {
+        position = getSurfacePos(position);
+        if ((position == null)
+            || field.isNoPartOfField(world, position) 
+            || (world.getBlockState(position.above()).getBlock() instanceof CropsBlock)
+            || (world.getBlockState(position.above()).getBlock() instanceof BlockScarecrow)
+            || !world.getBlockState(position).getBlock().is(Tags.Blocks.DIRT)
+        )
+            {
+                return null;
+            }
+        return position;
+    }
+
 
-        return !field.isNoPartOfField(world, position) && !(world.getBlockState(position.above()).getBlock() instanceof CropsBlock)
-                 && !(world.getBlockState(position.above()).getBlock() instanceof BlockScarecrow)
-                 && (world.getBlockState(position).getBlock().is(Tags.Blocks.DIRT) || world.getBlockState(position).getBlock() instanceof GrassBlock);
+    private BlockPos getSurfacePos(final BlockPos position)
+    {
+        return getSurfacePos(position, 0);
+    }
+    /**
+    * Finds the position of the surface near the specified position
+    *
+    * @param position the location to begin the search
+    * @return the position of the surface block or null if it can't be found
+    */
+    private BlockPos getSurfacePos(final BlockPos position, final Integer depth)
+    {
+        if (depth > MAX_DEPTH)
+        {
+            return position.below();
+        }
+        else if (depth < (MAX_DEPTH * -1))
+        {
+            return null;
+        }
+        final BlockState curBlockState = world.getBlockState(position);
+        @Nullable final Block curBlock = curBlockState.getBlock();
+        if (curBlock.is(Tags.Blocks.DIRT) || curBlock.is(Tags.Blocks.STONE) || curBlockState.getMaterial().isLiquid() || curBlock instanceof FarmlandBlock)
```

Comment:
I thought a bunch about this. We should start at position MAX_DEPTH and then move down to -MAX_DEPTH until we reach a "solid block". If this solid block is not farmland, we can't plant here. (Avoid planting in a little cave opening).

Reply:
The problem with that is what if you wanted your farmland in a cave? Or in a stone-ceilinged room? If we started at the top and worked down, it would prevent you from planting if the (stone or dirt) ceiling was lower than five blocks above

Reply:
Okay. So essentially we want to move in both direction in the same speed actually.
Like "until reaching MAX_DEPTH do i++
check pos.above(i) and pos.below(i) if it's farmland.

When down from the field, we want to stop once we hit a solid non field block
When going up, we want to stop once we reach an air block.

Reply:
I'm not sure I understand. Why would we want to check both directions? We're only looking for one surface block.

Reply:
To clarify: At the moment it still does check both up and down, but it figures out which direction the surface should be and only goes in that direction, rather than scanning both ways.

Reply:
We want to check if the farmland block is downwards or upwards. But we probably want to search in both directions in the same speed (as we return the first we find).

Reply:
The block might be solid but the next farmland block is actually down (because of the cave stuff)

Reply:
Yeah, it could be an issue if the land sloped down into a cave where the roof of the cave is at the same level as the block the scarecrow is sitting on. Turning down the MAX_DEPTH would make that much less likely, but either way it's an edge case that I can't see a good solution for. The only thing I could think of would be to track where the surface is, and start the search from the surface of an adjacent block. And that just seems messy.

Reply:
Also, I realized you mentioned looking for the farmland block, but this code is also used when the Farmer is determining where to hoe, before any farmland blocks exist.

Reply:
yeah, "farmland or block that can be turned into farmland" essentially.

Reply:
Okay, yeah. So the way the code works, there are some pretty rare situations where it could behave oddly, but I can't think of a way to easily fix it that wouldn't lead to it behaving oddly in more common situations.

Reply:
So are you good with this as it stands?